### PR TITLE
[HVG-136] Chore / replace tslint with eslint

### PR DESCRIPTION
**Pull request by styrelius**
### What?
Migrate from TSLint to ESLint. This entails:
1) Removing TSLint and all related plugins, config packages, the TSLint config files and all the `tslint:disable-next-line` and `tslint:disable` comments (in some places replacing them with `eslint-disable-next-line`).
2) Installing ESLint and the `@typescript-eslint/parser` and all the plugins and configs needed.
3) Adding an `.eslintrc.js` file with config and rules stuff and a `lint` script in `package.json`.
4) Attempting to fix all the new fun linting errors that occurred (most of them TypeScript related). 
In most cases the errors and warnings were fixable but in a few places I had to disable the `@typescript-eslint/ban-types` rule for lines using the `object` type since this type comes from `apollo-client`.

### Why?
TSLint is, as we all know, deprecated as of 2019 and the best option is ESLint in combination with [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint).

_Referenced ticket HVG-136 [here](https://hedvig.atlassian.net/browse/HVG-136)_

### CODE REVIEW

**Review by palmenhq — DISMISSED** [review]
Awesome!

File: src/client/pages/OfferNew/Introduction/Sidebar/index.tsx [hunk 1]
```diff
@@ -155,169 +155,166 @@ const FooterExtraActions = styled.div`
...
-export const Sidebar = React.forwardRef<HTMLDivElement, Props>(
-  ({ sticky, offerData, refetch, onCheckoutOpen }, ref) => {
-    const textKeys = useTextKeys()
-    const market = useMarket()
-    const [
-      discountCodeModalIsOpen,
-      setDiscountCodeModalIsOpen,
-    ] = React.useState(false)
-    const [isSidebarVisible, setIsSidebarVisible] = React.useState(true)
+export const Sidebar = React.forwardRef<HTMLDivElement, Props>(function sidebar(
```
Comment (gustaveen) [inline src/client/pages/OfferNew/Introduction/Sidebar/index.tsx:158]: I kinda dislike this rule and would rather add `'react/display-name': 'off',` what do you think? @styrelius @palmenhq

**Comment by gustaveen** [comment]
Great job with this one! 👌

File: src/client/pages/OfferNew/Introduction/Sidebar/index.tsx (hunk 1 above)
Comment (palmenhq) [reply src/client/pages/OfferNew/Introduction/Sidebar/index.tsx:158]: Sure!

Comment (styrelius) [reply src/client/pages/OfferNew/Introduction/Sidebar/index.tsx:158]: I have no opinion whatsoever in this matter so I can definitely turn that rule off and change those now named functions back into anonymous functions!

**Review by gustaveen — APPROVED** [review]
🥳

**Pull request merged**